server/src/main/java/org/elasticsearch/action/search/FetchSearchPhase.java

```diff
@@ -128,7 +127,7 @@ private void innerRun() throws Exception {
             finishPhase.run();
         } else {
             ScoreDoc[] scoreDocs = reducedQueryPhase.sortedTopDocs().scoreDocs();
-            final IntArrayList[] docIdsToLoad = searchPhaseController.fillDocIdsToLoad(numShards, scoreDocs);
+            final ArrayList<Integer>[] docIdsToLoad = searchPhaseController.fillDocIdsToLoad(numShards, scoreDocs);
```

Comment:
We should use List, not ArrayList, whenever passing these arround (ie in return types and local variables).